x-pack/plugins/fleet/server/services/package_policies/simplified_package_policy_helper.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { packageToPackagePolicy } from '../../../common/services';
+import type {
+  NewPackagePolicyInput,
+  NewPackagePolicyInputStream,
+  PackagePolicyConfigRecord,
+} from '../../../common/types';
+import type { NewPackagePolicy, PackageInfo } from '../../types';
+
+type SimplifiedVars = Record<string, string | string[] | boolean | number | number[] | null>;
+
+export interface SimplifiedPackagePolicy {
+  id?: string;
+  policy_id: string;
+  namespace: string;
+  name: string;
+  description?: string;
+  vars?: SimplifiedVars;
+  inputs?: Record<
+    string,
+    {
+      enabled?: boolean | undefined;
+      vars?: SimplifiedVars;
+      streams?: Record<
+        string,
+        {
+          enabled?: undefined | boolean;
+          vars?: SimplifiedVars;
+        }
+      >;
+    }
+  >;
+}
+
+export function generateInputId(input: NewPackagePolicyInput) {
+  return `${input.policy_template ? `${input.policy_template}-` : ''}${input.type}`;
+}
+
+function assignVariables(
+  userProvidedVars: SimplifiedVars,
+  varsRecord?: PackagePolicyConfigRecord,
+  ctxMessage = ''
+) {
+  Object.entries(userProvidedVars).forEach(([varKey, varValue]) => {
+    if (!varsRecord || !varsRecord[varKey]) {
+      throw new Error(`Variable ${ctxMessage}:${varKey} not found`);
+    }
+
+    varsRecord[varKey].value = varValue;
+  });
+}
+
+type StreamsMap = Map<string, NewPackagePolicyInputStream>;
+type InputMap = Map<string, { input: NewPackagePolicyInput; streams: StreamsMap }>;
+
+/**
+ *
+ * @param data
+ * @param packageInfo
+ */
+export function simplifiedPackagePolicytoNewPackagePolicy(
+  data: SimplifiedPackagePolicy,
+  packageInfo: PackageInfo
+): NewPackagePolicy {
+  const {
+    policy_id: policyId,
+    namespace,
+    name,
+    description,
+    inputs = {},
+    vars: packageLevelVars,
+  } = data;
+  const packagePolicy = packageToPackagePolicy(packageInfo, policyId, namespace, name, description);
+
+  // Build a input and streams Map to easily find package policy stream
+  const inputMap: InputMap = new Map();
+  // disable everything or use default?
```

Comment:
Is this an open question or something about the code below?

Reply:
Forget to clean that I decided to use the package default here, as I think it's the easier for the user, and it allow to get the package working with almost nothing provided by the user.